irksome/dirk_stepper.py

```diff
@@ -180,23 +184,27 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
             self.problem, appctx=appctx, solver_parameters=solver_parameters,
             nullspace=nullspace)
 
-        self.kgac = k, g, a, c
+        self.kgac = k, g, a, c, a_vals, d_val
+
+    def update_bc_constants(self, AAb, CCone, i, a_vals, d_val, c):
```

Comment:
Almost every argument to this function (except `i`) could be read directly from `self`, so there's no need to pass them as arguments.